stacks-node/src/burnchains/rpc/bitcoin_rpc_client/mod.rs

```diff
@@ -0,0 +1,510 @@
+        Ok(self
+            .global_ep
+            .send(&self.client_id, "listwallets", vec![])?)
+    }
+
+    /// Retrieve a list of unspent transaction outputs (UTXOs) that meet the specified criteria.
+    ///
+    /// # Arguments
+    /// * `min_confirmations` - Minimum number of confirmations required for a UTXO to be included (Default: 0).
+    /// * `max_confirmations` - Maximum number of confirmations allowed (Default: 9.999.999).
+    /// * `addresses` - Optional list of addresses to filter UTXOs by (Default: no filtering).
+    /// * `include_unsafe` - Whether to include UTXOs from unconfirmed unsafe transactions (Default: `true`).
+    /// * `minimum_amount` - Minimum amount (in BTC. As String to preserve full precision) a UTXO must have to be included (Default: "0").
+    /// * `maximum_count` - Maximum number of UTXOs to return. Use `None` for effectively unlimited (Default: 9.999.999).
+    ///
+    /// # Returns
+    /// A Vec<[`ListUnspentResponse`]> containing the matching UTXOs.
+    ///
+    /// # Notes
+    /// This method supports a subset of available RPC arguments to match current usage.
+    /// Additional parameters can be added in the future as needed.
+    pub fn list_unspent(
+        &self,
+        min_confirmations: Option<u64>,
+        max_confirmations: Option<u64>,
+        addresses: Option<&[&str]>,
+        include_unsafe: Option<bool>,
+        minimum_amount: Option<&str>,
+        maximum_count: Option<u64>,
+    ) -> BitcoinRpcClientResult<Vec<ListUnspentResponse>> {
+        let min_confirmations = min_confirmations.unwrap_or(0);
+        let max_confirmations = max_confirmations.unwrap_or(9999999);
+        let addresses = addresses.unwrap_or(&[]);
+        let include_unsafe = include_unsafe.unwrap_or(true);
+        let minimum_amount = minimum_amount.unwrap_or("0");
+        let maximum_count = maximum_count.unwrap_or(9999999);
+
+        Ok(self.wallet_ep.send(
+            &self.client_id,
+            "listunspent",
+            vec![
+                min_confirmations.into(),
+                max_confirmations.into(),
+                addresses.into(),
+                include_unsafe.into(),
+                json!({
+                    "minimumAmount": minimum_amount,
+                    "maximumCount": maximum_count
+                }),
+            ],
+        )?)
+    }
+
+    /// Mines a specified number of blocks and sends the block rewards to a given address.
+    ///
+    /// # Arguments
+    /// * `num_block` - The number of blocks to mine.
+    /// * `address` - The Bitcoin address to receive the block rewards.
+    ///
+    /// # Returns
+    /// A vector of block hashes corresponding to the newly generated blocks.
+    ///
+    /// # Availability
+    /// - **Since**: Bitcoin Core **v0.17.0**.
+    ///
+    /// # Notes
+    /// Typically used on `regtest` or test networks.
+    pub fn generate_to_address(
+        &self,
+        num_block: u64,
+        address: &str,
```

Comment:
Please use `stackslib::burnchains::bitcoin::address::BitcoinAddress` instead of `&str`.